sympy/polys/factortools.py

```diff
@@ -144,20 +150,20 @@ def dup_zz_hensel_step(m, f, g, h, s, t, K):
     Given positive integer `m` and `Z[x]` polynomials `f`, `g`, `h`, `s`
     and `t` such that::
 
-        f == g*h (mod m)
-        s*g + t*h == 1 (mod m)
+        f = g*h (mod m)
+        s*g + t*h = 1 (mod m)
 
         lc(f) is not a zero divisor (mod m)
-        lc(h) == 1
+        lc(h) = 1
 
-        deg(f) == deg(g) + deg(h)
+        deg(f) = deg(g) + deg(h)
         deg(s) < deg(h)
         deg(t) < deg(g)
 
     returns polynomials `G`, `H`, `S` and `T`, such that::
 
-        f == G*H (mod m**2)
-        S*G + T**H == 1 (mod m**2)
```

Comment:
should this be `T^H`?

Reply:
Hello, thanks for your review.

The line `S*G + T**H == 1 (mod m**2)` is contained in the document of the function `dup_zz_hensel_step`.

Here is the related testing function `test_dup_zz_hensel_step` in the module [`/sympy/polys/tests/test_factortools.py`](https://github.com/sympy/sympy/blob/master/sympy/polys/tests/test_factortools.py):
```
def test_dup_zz_hensel_step():
    R, x = ring("x", ZZ)

    f = x**4 - 1
    g = x**3 + 2*x**2 - x - 2
    h = x - 2
    s = -2
    t = 2*x**2 - 2*x - 1

    G, H, S, T = R.dup_zz_hensel_step(5, f, g, h, s, t)

    assert G == x**3 + 7*x**2 - x - 7
    assert H == x - 7
    assert S == 8
    assert T == -8*x**2 - 12*x - 1
```
We can see that an example is given. Let's take the conclusions after `assert` as conditions.

- Assuming that `S*G + T*H = 1 (mod m**2)` is correct:
```
>>> S*G + T*H
8*x**3 + 56*x**2 - 8*x + (x - 7)*(-8*x**2 - 12*x - 1) - 56

>>> factor(S*G + T*H - 1)
25*(4*x**2 + 3*x - 2)
```
`S*G + T*H - 1` has a factor `25`, the `m**2`, so `S*G + T*H = 1 (mod m**2)` must be established.

- Assuming that `S*G + T**H = 1 (mod m**2)` is correct:
```
>>> S*G + T**H
8*x**3 + 56*x**2 - 8*x + (-8*x**2 - 12*x - 1)**(x - 7) - 56

>>> factor(S*G + T**H - 1)
8*x**3 + 56*x**2 - 8*x + (-8*x**2 - 12*x - 1)**x/(-2097152*x**14 - 22020096*x**13 - 100925440*x**12 - 264241152*x**11 - 434208768*x**10 - 463454208*x**9 - 322187264*x**8 - 143523840*x**7 - 40273408*x**6 - 7241472*x**5 - 848064*x**4 - 64512*x**3 - 3080*x**2 - 84*x - 1) - 57
```
Compared with the former result, this one is obviously weird, and can't make `S*G + T**H = 1 (mod m**2)` established.

By the way, in the case of `A**B`, with respect to this part of mathematical logic, `A` is usually a polynomial and `B` is usually a constant, a specific integer. The problem might be over complex when `B` became a polynomial.

From all the above, I think `S*G + T**H == 1 (mod m**2)` is a careless miastake, the correct one is `S*G + T*H == 1 (mod m**2)`.

If there is still something I didn't explain clearly, please reply and let me know.